Refactor: Added ResultsList web layout

JoseDuSV opened the pull request:
Summary: Updated `ResultsList` and `ItineraryCard` web-related styles and conditions.

`TripSector` web layout will be updated in next PR.

@jvaclavik Do you have any plans with `ItineraryCardRow` component? As you can see I just updated its styles but I'm not sure if this wrapper is really necessary. This padding can probably be directly part of the `TripSector` component.

Related to #112

jvaclavik commented:
> Summary: Updated `ResultsList` and `ItineraryCard` web-related styles and conditions.
> 
> `TripSector` web layout will be updated in next PR.
> 
> @jvaclavik Do you have any plans with `ItineraryCardRow` component? As you can see I just updated its styles but I'm not sure if this wrapper is really necessary. This padding can probably be directly part of the `TripSector` component.

I don't have any special plans with `ItineraryCardRow`, so feel free to delete it.

jvaclavik commented on apps/core/components/ItineraryCard/ItineraryCard.js:
```diff
@@ -107,5 +122,35 @@ const styles = StyleSheet.create({
+      marginBottom: parseInt(defaultTokens.spaceMedium, 10),
+      borderBottomWidth: 0,
+      borderTopWidth: 0,
+      borderRadius: 3,
```
You can use `borderRadiusNormal` from `designTokens`

JoseDuSV replied:
Thanks, I thought that there is only 2 and 6. Updated

JoseDuSV commented:
> > Summary: Updated `ResultsList` and `ItineraryCard` web-related styles and conditions.
> > `TripSector` web layout will be updated in next PR.
> > @jvaclavik Do you have any plans with `ItineraryCardRow` component? As you can see I just updated its styles but I'm not sure if this wrapper is really necessary. This padding can probably be directly part of the `TripSector` component.
> 
> I don't have any special plans with `ItineraryCardRow`, so feel free to delete it.

Ok, thanks. I'll remove it as part of next PR related to `TripSector` web version.

jvaclavik commented on apps/core/components/ItineraryCard/ItineraryCard.js:
```diff
@@ -107,5 +122,35 @@ const styles = StyleSheet.create({
+  },
+  price: {
+    web: {
+      color: '#2e353b',
```
This color is not in orbit, but maybe it should be there. What do you think? If you agree we could leave here some TODO comment

JoseDuSV replied:
Ok, I just replaced this component with `LocalizedPrice` component so I also removed this style. But I'll be updating `LocalizedPrice` with this style so I'll include this comment there.

RobinCsl commented on apps/core/components/ItineraryCard/ItineraryCardRow.js:
```diff
@@ -17,5 +17,9 @@ const styles = StyleSheet.create({
     paddingHorizontal: 15,
     paddingVertical: 5,
+    web: {
+      paddingHorizontal: 0,
```
`padding:0` instead?

JoseDuSV replied:
I've tried it but it will not work. The values needs to be overridden directly like this.
I encounter same situation for example even in `ItineraryCard` with `borderBottomWidth` where setting global `borderWidth` to reset border wasn't enough.

RobinCsl commented on apps/core/components/ItineraryCard/ItineraryCard.js:
```diff
@@ -63,18 +69,27 @@ class ItineraryCard extends React.Component<Props> {
       <View style={styles.card}>
-        <View>
+        {Platform.OS === 'web' && (
+          <Text style={styles.price}>{formatPrice(priceObject)}</Text>
```
Should we instead change the `LocalizedPrice` component in @kiwicom/universal-components?

JoseDuSV replied:
I've planned to replace it once the `LocalizedPrice` web style is updated. But you are right, the current design is not that different so I replaced it with `LocalizedPrice` component and will update its web styles in `universal-components`.

RobinCsl commented:
Where is the design for this new web layout?

JoseDuSV commented:
> Where is the design for this new web layout?

Sorry, I should mention it in PR description (now updated) that it is related to #112 .
The design should be as close as possible to current kiwi.com results list.
